### ISSUE TITLE
Cursor renders out of place when pum opens/closes

### ISSUE THREAD

**Issue by antaed** [neoclide/coc.nvim#404]
![cursorjump](https://user-images.githubusercontent.com/30736986/52237560-84564400-28d2-11e9-8cc3-e3bb774de011.gif)

**Comment by chemzqm** [comment]
Please provide `:CocInfo`

**Comment by antaed** [comment]
Hmm, I only have this line in coc-info:
`## versions`

**Comment by chemzqm** [comment]
Could you try `set completopt-=preview` in your `.vimrc`?

**Comment by chemzqm** [comment]
What result from `:version` command?

**Comment by antaed** [comment]
`set completeopt-=preview` doesn't help
Here is the output of `:version`:
[version.txt](https://github.com/neoclide/coc.nvim/files/2829901/version.txt)
The issue also shows up on MacVim.

**Comment by chemzqm** [comment]
Ok, got the problem, I think I have to disable echo detail message on gui vim and try to use balloon instead.

**Comment by antaed** [comment]
Hi, thanks for the update but I'm sorry to report it doesn't seem to work for me.

I updated the plugin with `:call minpac#update()` then ran `:call coc#util#build()`. 
On MacVim the issue persists with or without `set completeopt-=preview` in vimrc.
On Windows the plugin doesn't work anymore, I get this message when I start gVim: `coc.nvim require python or python3 on vim`.
Did I do something wrong?

**Comment by chemzqm** [comment]
Upgrade macvim to latest version.
Compile gvim with python support,or you will get more bugs.

**Comment by chemzqm** [comment]
If you still have problem, paste your `:CocInfo` and provide a minimal `.vimrc` that could reproduce the problem.

**Comment by antaed** [comment]
The Vim build I use has Python support so I just installed Python 3.6 and the plugin works again.
I managed to narrow down my issues with the cursor to this code I have in my vimrc:

`inoremap <silent><expr> <C-j>`
`      \ pumvisible() ? "\<C-n>" :`
`      \ <SID>check_back_space() ? "\<C-j>" :`
`      \ coc#refresh()`
`inoremap <expr><C-k> pumvisible() ? "\<C-p>" : "\<C-h>"`
`function! s:check_back_space() abort`
`  let col = col('.') - 1`
`  return !col || getline('.')[col - 1]  =~# '\s'`
`endfunction`

Basically, I modified the code in your configuration example so that I can use C-k and C-j to move up and down the list (instead of S-Tab and Tab). Unfortunately I have no vimscript knowledge, can you please tell me what is wrong? Thanks

**Comment by chemzqm** [comment]
`<c-k>` is used for jump placeholder, checkout help to configure to other key.

**Comment by antaed** [comment]
Already did that with:
`let g:coc_snippet_next = '<Tab>'`
`let g:coc_snippet_prev = '<S-Tab>'`

I went back to the original mappings from your example but the issue still doesn't go away. 
I also mapped key combinations the plugin doesn't seem to use (C-q / C-d) just to rule out any possible clash. The issue persists also with these.
The only way I can prevent the cursor from jumping is if I use the arrow keys to navigate.

**Comment by chemzqm** [comment]
Try add `  "coc.preferences.diagnostic.enable": false,` to your coc-settings.json, if the issue not exists, you should upgrade your macvim.

**Comment by antaed** [comment]
Tried both suggestions but the cursor still jumps. Also, I observed the cursor jumping to the command line also in other contexts (eg. when opening a php file, right after the `[coc-nvim] Indexed php files` message is displayed)

**Comment by chemzqm** [comment]
@antaed What‘s result of `echo split(execute('version'),"\n")[2]` in your macvim?

**Comment by antaed** [comment]
Included patches: 1-873

**Comment by chemzqm** [comment]
That's same version of mine, maybe try upgrade coc.nvim to latest master again.

**Comment by chemzqm** [comment]
Could be caused by your autocmds, consider try it with a minimal vimrc.

**Comment by antaed** [comment]
Sorry, I've been afk but in the next few days I'll try to rebuild my vimrc step-by-step to until I find the culprit for my issues

**Comment by antaed** [comment]
happy to report my issues are gone after updating and rebuilding coc.nvim and then updating extensions :)
